architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+
+4. **Communication with UCPD**: Controller also calls UCPD when doing reconcilations. Here is the list of instances where Controller calls UCPD:
+
+   1. Recipe Reconciler:
+
+      - Poll operations like create, update, or delete a Recipe resource.
+      - Resolve dependencies for a Recipe resource:
+        - Get the Radius environment the Recipe is associated with.
+        - Create the Radius Resource Group if needed.
+        - Create the Radius Application if needed.
+      - Fetch a Radius resource.
+      - List Secrets of a Recipe resource.
+      - Delete a Recipe resource.
+
+   2. Deployment Reconciler:
+
+      - Poll operations like create, update, or delete a Deployment resource.
+      - Resolve dependencies for a Deployment resource:
+        - Get the Radius environment the Deployment is associated with.
+        - Create the Radius Resource Group if needed.
+        - Create the Radius Application if needed.
+      - Fetch a Recipe resource.
+      - List Secrets of a Recipe resource.
+
+5. **Communication with Kubernetes API Server**: Controller also calls the Kubernetes API Server when doing reconcilations. Here is the list of instances where Controller calls the Kubernetes API Server:
+
+   1. Recipe Reconciler:
+
+      - Fetch a Recipe object.
+      - Send events related to the operations.
+      - Update a Recipe object:
+        - Status update.
+        - Addition and removal of a Finalizer.
+      - Delete a Secret object that is associated with the Recipe.
+      - Create a Secret object to associate with the Recipe if needed.
+
+   2. Deployment Reconciler:
+
+      - Fetch a Deployment object.
+      - Send events related to the operations.
+      - Update a Deployment object:
+        - Status update.
+        - Addition and removal of a Finalizer.
+      - Delete a Secret object associated with a Deployment that is no longer needed.
+      - Fetch a Secret object.
+      - Create or Update a Secret object and associate it with a Deployment object.
+      - List Deployments filtered with specific Recipe objects.
+
+### Threats
+
+#### Impersonation of Radius Controllers to trigger resource access
+
+Risk: An attacker can impersonate the Radius Controllers to talk to UCPD.
+
+Impact: The Controller Component and UCPD communicate in order to create/update/delete necessary resources and these resources can also be cloud resources. This can lead to loss of resources or creation/update of existing resources.
+
+Mitigation:
+
+1. **mTLS**: mTLS should be (and is) enabled for communication between the Controller and the UCPD.
+2. **Network Policies**: We can add necessary policies that UCPD can only accept communication from the Controller and other important components that it is should be talking to.
+
+#### Compromised TLS Certificates and private keys of the Validating Webhook
```

Comment:
You need to register as the validating webhook so that Kube API Server can send you the requests.

If it is admin doing something, then it is admin, and they can do what they want.

If not admin, then can they act as the Validating Webhook?